### PR TITLE
add pipework catalog

### PR DESCRIPTION
**Pull request by mittz**
*Software-Defined Networking for Linux Containers*

* GitHub: [jpetazzo/pipework](https://github.com/jpetazzo/pipework)
* Docker Hub: [dreamcat4/pipework](https://hub.docker.com/r/dreamcat4/pipework/)

By this catalog, to configure more flexible container network is available.  
A good example of this, there is multiple container networks.

### CODE REVIEW

File: infra-templates/pipework/0/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    stdin_open: true
+    network_mode: host
+    volumes:
+    - /var/run/docker.sock:/docker.sock
+    tty: true
+    pid: host
+    environment:
+      run_mode: batch,daemon
+    labels:
+      io.rancher.container.pull_image: always
```
Comment (rawmind0) [inline infra-templates/pipework/0/docker-compose.yml]: You don't need to pull image always
```
-      io.rancher.container.pull_image: always
```

File: infra-templates/pipework/0/rancher-compose.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+.catalog:
```
Comment (rawmind0) [inline infra-templates/pipework/0/rancher-compose.yml]: Could you please use version 2 for whole file??
```
-.catalog:
+version: '2'
+catalog:
```

File: infra-templates/pipework/0/rancher-compose.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+  name: "Pipework"
+  description: |
+    "Software-Defined Networking for Linux Containers"
+  version: v0.1.0
+  minimum_rancher_version: v1.6.10
+version: '2'
```
Comment (rawmind0) [inline infra-templates/pipework/0/rancher-compose.yml]: Don't need version 2 here...
```
-version: '2'
```

**Comment by rawmind0** [comment]
Hi @mittz , thanks for the PR and to contribute.

Please, take a look to the changes requested.

**Comment by mittz** [comment]
Hi @rawmind0 , thank you for your comments.
I made some changes, could you check them?

**Comment by rawmind0** [comment]
@mittz thanks for the changes.

LGTM

**Pull request merged**